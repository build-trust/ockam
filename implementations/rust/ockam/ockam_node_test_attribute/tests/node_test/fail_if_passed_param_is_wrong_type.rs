// This test checks that #[ockam_node_test_attribute::node_test] causes a compile time error
// if the function is passed a param that is not of type `ockam::Context`

#[ockam_node_test_attribute::node_test]
async fn my_test(ctx: std::string::String) -> ockam::Result<()> {}
