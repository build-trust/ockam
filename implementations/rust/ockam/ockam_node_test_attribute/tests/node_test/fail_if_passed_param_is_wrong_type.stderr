error: Expected `ockam::Context` found `std::string::String`
 --> tests/node_test/fail_if_passed_param_is_wrong_type.rs:5:23
  |
5 | async fn my_test(ctx: std::string::String) -> ockam::Result<()> {}
  |                       ^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/node_test/fail_if_passed_param_is_wrong_type.rs:4:1
  |
4 | / #[ockam_node_test_attribute::node_test]
5 | | async fn my_test(ctx: std::string::String) -> ockam::Result<()> {}
  | |__________________________________________________________________^ consider adding a `main` function to `$DIR/tests/node_test/fail_if_passed_param_is_wrong_type.rs`
